### PR TITLE
Add Isaac Morales blog

### PR DESCRIPTION
**Pull request by imorlxs**
Project: [Improving performance of BioDynaMo using ROOT C++ Modules](https://hepsoftwarefoundation.org/gsoc/2024/proposal_ROOTBDM.html)

### CODE REVIEW

**Comment by vvolkl** [comment]
Very nice, thanks! Giving @vgvassilev @LukasBreitwieser a chance to comment before merging

**Review by vgvassilev — APPROVED** [review]
Lgtm!

**Review by LukasBreitwieser — CHANGES_REQUESTED** [review]
Hi @imorlxs ,
Thank you very much for your work and blog. I left a few comments to strengthen the text. Let me know if you have any questions.
Best,
Lukas

File: _gsocblogs/2024/blog_BioDynaMo_IsaacMorales.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+As expected, the simulation time did not show an appreciable improvement. However, in the
+unit tests, the time was 33% lower. I believe this is because unit tests involve a lot of parsing and Cling calls.
+
+### Future Steps and Challenges Ahead
+Despite these advances, several challenges remain. PR's #365 is ready to merge and #385 needs some changes. For instance, memory leaks have been observed when using the new
+`ROOT_GENERATE_DICTIONARY`, even with C++ Modules disabled. Additionally, the build system for individual demos has
+caused compatibility issues with the main build system. Also, there is a problem with the Jupyter notebooks.
+Resolving these issues and finalizing the integration of C++ Modules will be essential for ensuring long-term stability and reliability.
+
+Looking ahead, further optimizations are planned, including potential module-based optimizations for BioDynaMo’s
```
[the remark this answers is not in the document]
Comment (imorlxs) [reply _gsocblogs/2024/blog_BioDynaMo_IsaacMorales.md:112]: For now we are generating one big module (biodynamo.pcm). The goal is to split that into smaller ones to load them on-demand.

**Review by LukasBreitwieser — APPROVED** [review]

**Review by vvolkl — APPROVED** [review]

**Pull request merged**